scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java

```diff
@@ -109,7 +109,7 @@ public void register(NotificationRequest notificationRequest) throws Notificatio
             }
         }
         // All past events have been scheduled. Nothing to schedule in the future.
-        if (request.getEndTime().isBefore(nextStartTime)) {
+        if (endTime.isBefore(nextStartTime)) {
```

Comment:
request.getEndTime is also being used in creating trigger for the quartz job. Doesn't that need to be changed or is it already exclusive in quartz?